common/lib/common-utils/src/idleTaskScheduler.ts

```diff
@@ -0,0 +1,35 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+/**
+ * API used to schedule non-essential tasks
+ * Time measurements are in milliseconds as a floating point with a decimal
+ * Takes in and runs a callback during idle time. Fallback to setTimeout if window doesn't
+ * support requestIdleCallback.
+ * @returns A promise pertaining to the callback that was passed in.
+ */
+export async function scheduleIdleTask<T>(callback: () => T, timeout: number): Promise<T> {
+    // Check for the availability
+    return typeof globalThis.requestIdleCallback === "function"
+        ? new Promise<T>((resolve, reject) => {
+              const doLowPriorityTask = (): void => {
+                  try {
+                      resolve(callback());
+                  } catch (err: any) {
+                      reject(err);
+                  }
+              };
+              requestIdleCallback(doLowPriorityTask, { timeout });
+          })
+        : new Promise<T>((resolve, reject) => {
+              setTimeout(() => {
```

Comment:
I feel like I've left that feedback 3 times now :)
Move definition of doLowPriorityTask up a bit and use it here with timer as well - no need to duplicate code - what you pass to setTimeout is exactly what doLowPriorityTask  is.